cigaradvisor/scripts/scripts.js

```diff
@@ -229,6 +229,16 @@ export async function getPostByIdx(idx) {
   return undefined;
 }
 
+/**
+ * Retrieves all posts and sorts them by published date.
+ * @param {boolean} sort - Indicates whether to sort the posts by published date in ascending order.
+ * @returns {Promise<Array>} - A promise that resolves to an array of post data.
+ */
+export async function getAllPosts() {
+  await loadPosts();
+  return articleIndexData;
```

Comment:
I'm now realizing that relying on the variable to be populated by insuring to call `loadPosts` doesn't make sense.

Can you update `loadPosts` to return the stored post list, and change `getPostByIndex` and `fetchPostsInfo` to not use the "global" variable?

Reply:
fixed